### DISCUSSION TITLE
exitCode 1 == one test failed && exitCode 1 == error

### DISCUSSION THREAD

**Discussion by sebamomann** [cypress-io/cypress#24718] [Questions and Help]
Hi,

the cypress [documentation](https://docs.cypress.io/guides/guides/command-line#Exit-code) suggests that the exit code of the `cypress run` command reflects the number of failed tests. That's totally fine, unless you only have one failed test. This is because the second paragraph suggests that upon other errors (like a missing spec file) the status code 1 will be returned too. This results in confusion since i need to differenciate between a logical error inside the test (failed assertion etc) and technical failures (like a non existing spec file etc.)

Is there any possibility to not use the error code 1 for completely different outcomes?

**Comment by portyu9** [comment]
Hi @sebamomann 

Yes — **exit code `1` alone is not sufficient to distinguish those two cases**.

This is still true in current Cypress.

With the normal exit-code behavior:

- one failed test can produce `1`
- Cypress being unable to run can also produce `1`

`--posix-exit-codes` makes the behavior more conventional and gives some failures separate codes (for example certain required Cypress Cloud network failures), but `1` can still mean either test failures or a generic "Cypress could not run" condition.

So if your CI needs to classify the result, don't use the process exit code alone. Capture either:

- the result from the Cypress **Module API**, or
- structured reporter output such as JSON/JUnit

and inspect whether tests actually ran and how many failed.

Cypress's current CLI documentation specifically recommends the Module API or structured reporter data when you need the number of failed tests while using POSIX exit codes.

For the particular "no spec files found" case, current Cypress also has:

```bash
cypress run --pass-with-no-tests
```

if that condition should not fail your CI job.

Please mark this answer as accepted if it helped, thank you.
